### ISSUE TITLE
[Question] Are old sources retained when (git) packages are rebuilt?

### ISSUE THREAD

**Issue by xiota** [chaotic-aur/packages#2271]
Are old sources retained when packages are rebuilt?  What the best way to organize -git packages with submodules?

If old sources are retained, it might be worthwhile to track each git submodule in the PKGBUILD sources so they can be updated instead of redownloaded every time the package is rebuilt.

**Comment by dr460nf1r3** [comment]
Yeah, git sources are cached. Whenever the package gets updated, the toolbox pulls new commits. I'm actually not sure how it behaves concerning submodules though.

**Comment by xiota** [comment]
When the git source changes, is the system smart enough to clear the cache?  Or is manual intervention required?  For instance, what happens if it's changed from `"$pkgname"::"$url.git"` to `"${pckname#-git}"::"$url"` ?

Re submodules.  There's a package I'm interested in, but it has about 900MB of submodules that are redownloaded every time the package is rebuilt.  I commented with a request to restructure so that the submodules can be cached, but haven't heard back from the maintainer.  What would the convention for package naming be in this case?  Would this be worthwhile?  How consequential is 900MB per rebuild for chaotic aur?

**Comment by dr460nf1r3** [comment]
Good point - we actually need to manually clear the sources. Though [we are able to filter our logs](https://builds.garudalinux.org/repos/chaotic-aur/logs/filtered/source-changed/) to easily spot such packages.

As for the submodules - I'll come back to you soon.

**Comment by PedroHLC** [comment]
It's not the archway to use "git submodules". It's the archway to bare-clone the subrepos as secondary sources (by adding it to the PKGBUILD's sources array) and link+checkout them using "git config". **Bare-clones are correctly handled by our cache.**

See the official Arch docs about this: https://wiki.archlinux.org/title/VCS_package_guidelines#Git_submodules

**Comment by FabioLolix** [comment]
Submodules should be downloade in source=() and configured in prepare() otherwise will be downloaded every time that src is cleaned/deleted

**Comment by xiota** [comment]
@PedroHLC @FabioLolix  Yes, but what to do when the maintainer doesn't respond?

**Comment by PedroHLC** [comment]
> @PedroHLC @FabioLolix Yes, but what to do when the maintainer doesn't respond?

The submodule maintainer? Fork it and change the URL, the linear-timeline of git will keep the cache valid.

The PKGBUILD maintainer doesn't add it to `sources`? Send a message to [aur-general](https://lists.archlinux.org/mailman3/lists/aur-general.lists.archlinux.org/) asking the TUs to orphanage a non-complaining and abandoned PKGUILD. Linking your comment as proof of attempt and the wiki topic as the guideline they're disobeying.

**Comment by PedroHLC** [comment]
> @PedroHLC @FabioLolix Yes, but what to do when the maintainer doesn't respond?

In the worst-case scenario, we can append to the sources and overwrite the `prepare` of a PKGBUILD through this repo: https://github.com/chaotic-aur/interfere

**Comment by xiota** [comment]
@PedroHLC Thank you for your reponses.  I'd already written a comment, a couple weeks ago, on the aur package page with the request to add git submodules to sources.  I've edited the comment to add the [vcs packaging guidelines](https://wiki.archlinux.org/title/VCS_package_guidelines#Git_submodules) link.  I also flagged the package with the same info.  If there's no response or the maintainer unflags without updating the package, I'll proceed with your recommendation to request that the package be orphaned.
